security-analysis/src/main/java/eu/itesla_project/security/Security.java

```diff
@@ -28,75 +28,93 @@ public Security() {
 
-    private static float getCurrentLimit(CurrentLimits limits, CurrentLimitType limitType, int maxAcceptableDuration) {
-        float limit = Float.NaN;
-        switch (limitType) {
-            case PATL:
-                limit = limits.getPermanentLimit();
-                break;
-
-            case TATL:
-                for (CurrentLimits.TemporaryLimit tempoLimit : limits.getTemporaryLimits()) {
-                    if (tempoLimit.getAcceptableDuration() <= maxAcceptableDuration) {
-                        limit = tempoLimit.getLimit();
-                        break;
-                    }
-                }
-                break;
-
-            default:
-                throw new AssertionError();
-        }
-        return limit;
+    private static Country getCountry(TwoTerminalsConnectable branch, Terminal terminal) {
+        return terminal == branch.getTerminal1() ? branch.getTerminal1().getVoltageLevel().getSubstation().getCountry()
+                                                 : branch.getTerminal2().getVoltageLevel().getSubstation().getCountry();
     }
 
-    public static void checkCurrentLimits(TwoTerminalsConnectable branch, Terminal terminal, CurrentLimits limits,
-                                          CurrentLimitType limitType,int maxAcceptableDuration, float limitReduction,
-                                          List<LimitViolation> violations) {
-        if (limits != null) {
-            float i = terminal.getI();
-            if (!Float.isNaN(i)) {
-                float limit = getCurrentLimit(limits, limitType, maxAcceptableDuration);
-                if (!Float.isNaN(limit)) {
-                    if (i > limit * limitReduction) {
-                        Country country = terminal == branch.getTerminal1() ? branch.getTerminal1().getVoltageLevel().getSubstation().getCountry()
-                                : branch.getTerminal2().getVoltageLevel().getSubstation().getCountry();
-                        float baseVoltage = Math.max(branch.getTerminal1().getVoltageLevel().getNominalV(),
-                                                     branch.getTerminal2().getVoltageLevel().getNominalV());
-                        violations.add(new LimitViolation(branch, LimitViolationType.CURRENT, limit, limitReduction, i, country, baseVoltage));
-                    }
-                }
-            }
-        }
+    private static float getBaseVoltage(TwoTerminalsConnectable branch) {
+        return Math.max(branch.getTerminal1().getVoltageLevel().getNominalV(),
+                        branch.getTerminal2().getVoltageLevel().getNominalV());
     }
 
     public static void checkCurrentLimits(Iterable<? extends TwoTerminalsConnectable> branches, CurrentLimitType currentLimitType,
-                                          int maxAcceptableDuration, float limitReduction, List<LimitViolation> violations) {
+                                          float limitReduction, List<LimitViolation> violations) {
```

Comment:
I changed this method to private and added null checks to public methods checkLimits and printLimitsViolations